test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -50,11 +61,139 @@ public void AddAspNetCoreInstrumentation_BadArgs()
         Assert.Throws<ArgumentNullException>(() => builder.AddAspNetCoreInstrumentation());
     }
 
+#if NET8_0_OR_GREATER
     [Fact]
-    public async Task RequestMetricIsCaptured_Old()
+    public async Task ValidateNet8MetricsAsync()
+    {
+        var metricItems = new List<Metric>();
+
+        this.meterProvider = Sdk.CreateMeterProviderBuilder()
+            .AddAspNetCoreInstrumentation()
+            .AddInMemoryExporter(metricItems)
+            .Build();
+
+        var builder = WebApplication.CreateBuilder();
+        builder.Logging.ClearProviders();
+        var app = builder.Build();
+
+        app.MapGet("/", () => "Hello");
+
+        _ = app.RunAsync();
+
+        using var client = new HttpClient();
+        var res = await client.GetStringAsync("http://localhost:5000/").ConfigureAwait(false);
```

Comment:
If we really are expecting these tests to run in parallel (I don't think they actually do) we should probably randomize the port and handle the case where it is in use? I know we do that for other tests where we need to start up a listener.

@davidfowl Any other patterns perhaps we could use to cause the pieces of aspnetcore to fire we need instead of exercising the whole pipeline?

Reply:
Yes I think these tests need to be rewritten to use the WebApplicationFactory and fakes for the meter listener.

cc @JamesNK as he wrote tests for meters in ASP.NET Core.

Reply:
Randomizing port is something I was planning to do to avoid conflicts (we do have issues right now related to it).

we use WebApplicationFactory in most of the tests here. This new pattern is used as it relatively lightweight and does not require an additional template app like WebApplicationFactory does. If WebApplicationFactory is the recommended way then we can stick with it.

Reply:
same as https://github.com/open-telemetry/opentelemetry-dotnet/pull/4934#discussion_r1367416969